dynamic_stack_decider/src/dynamic_stack_decider/sequence_element.py

```diff
@@ -0,0 +1,58 @@
+from dynamic_stack_decider.abstract_stack_element import AbstractStackElement
+from dynamic_stack_decider.abstract_action_element import AbstractActionElement
+
+
+class SequenceElement(AbstractStackElement):
+    """
+    A sequence element contains multiple action elements.
+    The sequence element executes all the actions consecutively. That means that the first action will be performed
+    until it pops itself off the stack. Then, the second action is performed. When the last action is popped, the
+    sequence element pops itself too.
+    This is not an abstract class to inherit from.
+    """
+    def __init__(self, blackboard, dsd, actions=()):
+        """
+        :param actions: list of initialized action elements
+        """
+        super(SequenceElement, self).__init__(blackboard, dsd)
+        self.actions = actions
+        self.current_action_index = 0
+
+    def perform(self, reevaluate=False):
+        self.current_action.perform()
+
+    def pop_one(self):
+        """
+        Pop a single element of the sequence
+        :return: True when the pop was successful, False when the end of the sequence has been reached
+        """
+        self.current_action_index += 1
+        if self.current_action_index == len(self.actions):
+            return False
+        else:
+            return True
+
+    @property
+    def current_action(self):
+        """
+        :return: The currently executed action of the sequence element
+        :rtype: AbstractActionElement
+        """
+        return self.actions[self.current_action_index]
+
+    def repr_dict(self):
+        """
+        Represent this stack element as dictionary which is JSON encodable
+
+        :rtype: dict
+        """
+        self.clear_debug_data()
```

Comment:
Why is the debug data cleared before being sent?

I understand that you are setting it in the following lines but this is not an intuitive behavior. It would probably be better to clear it after sending.

Reply:
Done in 5925e7345efb4e2e3a6020652d53cbb3e9e75105